proposals/sycl_modules.md

```diff
@@ -707,66 +738,186 @@ value to the specialization constant.
 The program can then be build with `build_with_kernel_type` to create a 
 specialized kernel.
 
+### Specialization constant types
+
 On the new, type-safe, module approach, specialization constants are 
-associated with the OpenCL backend. 
-Other backends (e.g. Vulkan) may offer the same functionality, so the
-specialization constant type is defined as an (experimental) general 
-SYCL type as below:
+typically associated with SPIR-V module images but are not limited to such image.
+The specialization constant type is defined as an general SYCL type as below:
 
 ```cpp
 namespace sycl {
-namespace experimental {
 
-template <typename T, typename ID = T>
-class spec_constant {
+template <typename T>
+class spec_id {
```

Comment:
I reused the SPIR-V naming convention (`OpSpecConstant`) as in-fine this is to allow its representation in SYCL.

I'm fine to change `spec` to `specialization`. I don't really see a need, but you are in a better position to say if this makes things easier to understand.

Reply:
I made the change in this commit, https://github.com/KhronosGroup/SYCL-Shared/pull/1/commits/ad02463fe97933c35a10d5bccab75f53e991613b

I'm not convince we gain anything with this extra verbosity though.

Reply:
We gain a clarification between `specification` and `specialization` confusion.
Since this is a niche use case, I do not think it will jeopardize the programmer productivity but will help the reader not familiar with SPIR-V details...

Reply:
````
will help the reader not familiar with SPIR-V details
```

Maybe I should stop following SPIR-V then :)

If you feel it is better, we'lll stick to it, no strong opinion on this